ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+      alertText: (data, matches, output) => {
+        data.engravement1Towers.push(matches.target);
+        if (data.me === matches.target) {
+          if (matches.effectId === engravementIdMap.lightTower)
+            return output.lightTower!();
+          return output.darkTower!();
+        }
+      },
+      outputStrings: {
+        lightTower: {
+          en: 'Drop light tower',
+        },
+        darkTower: {
+          en: 'Drop dark tower',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Tower Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTiltIds },
+      condition: (data, matches) => data.engravementCounter === 1 && data.me === matches.target,
+      suppressSeconds: 5, // avoid second (incorrect) alert when debuff switches from soaking tower
+      alertText: (data, matches, output) => {
+        if (data.engravement1Towers.indexOf(data.me) === -1) {
+          // Did not drop a tower, so needs to soak one.
+          if (matches.effectId === engravementIdMap.lightTilt)
+            return output.lightTilt!();
+          return output.darkTilt!();
+        }
+      },
+      outputStrings: {
+        lightTilt: {
+          en: 'Soak dark tower',
+        },
+        darkTilt: {
+          en: 'Soak light tower',
+        },
+      },
+    },
+    // In Engravement 2 (Superchain 1), all supports or  DPS will receive lightTilt and darkTilt (2 each).
+    // All 4 also receive Heavensflame Soul.
+    // The other role group will receive lightTower, darkTower, lightBeam, and darkBeam.
+    // To resolve the Beams during the 2nd orb, lightBeam needs to stack with darkTower and both darkTilts, and vice versa.
+    // After the 3rd orb, lightTower and darkTower will drop their towers, and  darkBeam and lightBeam (respectively) will soak them.
+    // The four Heavensflame players all simultaneously need to spread to drop their AoEs.
+    // Debuffs do change based on mechanic resolution, which can complicate things:
+    // - When a lightTilt player soaks a dark beam, their debuff will change to darkTilt, and vice versa.
+    // - Once the beams detonate, the lightBeam debuff disappears and is replaced with lightTilt (same with dark).
+    // So only use the initial debuff to resolve the mechanic, and use a long suppress to avoid incorrect later alerts.
+    {
+      id: 'P12S Engravement 2 Beam Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: Object.values(engravementIdMap) },
+      condition: (data, matches) => data.engravementCounter === 2 && data.me === matches.target,
+      delaySeconds: 5, // wait until first orb resolves so as not to confuse
+      durationSeconds: 7, // keep active until beams go off
+      suppressSeconds: 30,
+      response: (_data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          lightBeam: {
+            en: 'Light Beam (stack with Dark group)',
```

Comment:
I think the "stack with X" is probably unnecessary here? I think this is a "people need to know where to go".  If it was always "dark group left" or something, then I think that'd make sense to say, but when it's just "stack with dark group, wherever that is", I think it's not helpful; folks already have to know the mechanic.

Reply:
> I think the "stack with X" is probably unnecessary here? I think this is a "people need to know where to go". If it was always "dark group left" or something, then I think that'd make sense to say, but when it's just "stack with dark group, wherever that is", I think it's not helpful; folks already have to know the mechanic.

I think I agree with that on the lightTower and darkTower alerts (i.e., I think we can remove '(stack with light/dark group)' from the alert).  For the others though:
- lightTilt and darkTilt just say 'Stack (light group)' or 'Stack (dark group)' - since we don't otherwise call out the color of their debuff, I think we should keep those parentheticals.
- For lightBeam and darkBeam, I do think it's worthwhile keeping '(stack with Dark/Light group)' - there's a lot going on during the mechanic, and I've found that people sometimes see the light debuff, forget that it's beam, and go stack with the light group (=wipe).  I think similar things could happen if we just say "Light Beam" -- people might see the light and forget the important part, e.g. that it's Beam and that they need to stack with the other group.

Reply:
Ok, if you think it's worth keeping the parenthetical on top of the different severity, sounds good to me.  Maybe drop the `group` part and just say `(stack with dark)` or `(stack with light)` for more brevity?